src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4231,7 +4231,7 @@ public bool ConnectAsync(SocketAsyncEventArgs e)
                     {
                         InternalBind(new IPEndPoint(IPAddress.Any, 0));
                     }
-                    else
+                    else if (endPointSnapshot.AddressFamily == AddressFamily.InterNetworkV6)
```

Comment:
So, are you saying that Desktop-based implementation must be also fixed to check for the address family here? But that's a different repo, and it must be a different work item? I can open a bug there, what repo is this?

As for the other socket types, without the changes from the pull request, only IPv4, v6 and DNS would work. So now other families can work too with custom EndPoints. Are you proposing to add a test for yet another family (not IPv4, v6, DNS, or domain socket)? And not only that, it must have a corresponding test in Desktop? Are there any existing tests for sockets that are not ip v4, v6, or DNS on Desktop? If Desktop has the same issue, then these kind of sockets won't work there too now.